cpp/include/copc-lib/io/laz_writer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef COPCLIB_IO_LAZ_WRITER_H_
+#define COPCLIB_IO_LAZ_WRITER_H_
+
+#include <array>
+#include <memory>
+#include <optional>
+#include <ostream>
+#include <stdexcept>
+#include <string>
+
+#include "copc-lib/copc/copc_config.hpp"
+#include "copc-lib/geometry/vector3.hpp"
+#include "copc-lib/io/laz_base_writer.hpp"
+#include "copc-lib/las/header.hpp"
+#include "copc-lib/las/laz_config.hpp"
+#include "copc-lib/las/points.hpp"
+#include "copc-lib/las/utils.hpp"
+
+namespace copc::laz
+{
+
+class LazWriter : public BaseWriter
+{
+
+  public:
+    LazWriter(std::ostream &out_stream, const las::LazConfigWriter &las_config);
+
+    // Write a group of points as a chunk
+    void WritePoints(const las::Points &points);
+
+    std::shared_ptr<las::LazConfigWriter> LazConfig() { return config_; }
+    uint64_t PointCount() { return point_count_; }
+    uint64_t ChunkCount() { return chunks_.size(); }
+
+  private:
+    int32_t WriteChunk(std::vector<char> in, int32_t point_count, bool compressed);
+};
+
+class LazFileWriter
+{
+
+  public:
+    LazFileWriter(const std::string &file_path, const las::LazConfigWriter &laz_config_writer);
```

Comment:
Should this be subclassed from LazWriter? Or do you want to keep them seperated?

Reply:
I only worry that it increases code duplication between the two, so it's easier to get out of sync

Reply:
Only problem is that we then have to override the Close function, and since it's called in the destructor and calling virtual functions in destructor is apparently not recommended. I agree though that it would be best to reduce code duplicate. Maybe we can chat about a way to avoid virtual call in destructor.